// jsii/build/deprecation-removal.js
"use strict";
var _a, _b, _c;
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterfaceFactory = exports.DeprecatedImplementation = exports.VisibleBaseClass = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
class VisibleBaseClass {
    constructor() {
        this.propertyPresent = true;
    }
}
exports.VisibleBaseClass = VisibleBaseClass;
_a = JSII_RTTI_SYMBOL_1;
VisibleBaseClass[_a] = { fqn: "@scope/jsii-calc-lib.deprecationRemoval.VisibleBaseClass", version: "0.0.0" };
/** @deprecated do not use me! */
class DeprecatedImplementation extends VisibleBaseClass {
    method() {
        /** NOOP */
    }
}
exports.DeprecatedImplementation = DeprecatedImplementation;
_b = JSII_RTTI_SYMBOL_1;
DeprecatedImplementation[_b] = { fqn: "@scope/jsii-calc-lib.deprecationRemoval.DeprecatedImplementation", version: "0.0.0" };
class InterfaceFactory {
    constructor() { }
    static create() {
        return new DeprecatedImplementation();
    }
}
exports.InterfaceFactory = InterfaceFactory;
_c = JSII_RTTI_SYMBOL_1;
InterfaceFactory[_c] = { fqn: "@scope/jsii-calc-lib.deprecationRemoval.InterfaceFactory", version: "0.0.0" };
